modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/AbstractXmlResourceFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.capability.xml.schema;
+
+import static org.mule.runtime.module.extension.internal.capability.xml.schema.model.SchemaConstants.XSD_EXTENSION;
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.property.XmlModelProperty;
+import org.mule.runtime.extension.api.resources.GeneratedResource;
+import org.mule.runtime.extension.api.resources.spi.GeneratedResourceFactory;
+
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Base class for {@link GeneratedResourceFactory} implementations which
+ * generate resources needed to support configuring extensions through
+ * XML
+ *
+ * @since 4.0
+ */
+abstract class AbstractXmlResourceFactory implements GeneratedResourceFactory
+{
+
+    /**
+     * Tests the given {@code extensionModel} to be enriched with the {@link XmlModelProperty}.
+     * If the property is present, then it delegates into {@link #generateXmlResource(ExtensionModel, XmlModelProperty)}.
+     * <p>
+     * Otherwise, it returns {@link Optional#empty()}
+     *
+     * @param extensionModel the {@link ExtensionModel} that requires the resource
+     * @return an {@link Optional} {@link GeneratedResource}
+     */
+    @Override
+    public Optional<GeneratedResource> generateResource(ExtensionModel extensionModel)
+    {
+        XmlModelProperty xmlProperty = extensionModel.getModelProperty(XmlModelProperty.class).orElse(null);
+
+        return xmlProperty == null
+               ? Optional.empty()
+               : Optional.of(generateXmlResource(extensionModel, xmlProperty));
+    }
+
+    /**
+     * Delegate method which should contain the actual logic to generate the resource
+     *
+     * @param extensionModel   the {@link ExtensionModel} that requires the resource
+     * @param xmlModelProperty the extension's {@link XmlModelProperty}
+     * @return a {@link GeneratedResource}
+     */
+    protected abstract GeneratedResource generateXmlResource(ExtensionModel extensionModel, XmlModelProperty xmlModelProperty);
+
+    /**
+     * Colon is a special character for the {@link Properties} class
```

Comment:
Too much detail. I would write "Escapes special characters for the {@link Properties} class that Spring uses to parse the bundle."